qiskit_metal/draw/utility.py

```diff
@@ -609,3 +610,250 @@ def snap_unit_vector(vec_n: Vec2D, flip: bool = False) -> Vec2D:
+        """
+        if not isinstance(vec, np.ndarray):
+            vec = np.array(vec)
+        if not isinstance(cvec, np.ndarray):
+            cvec = np.array(cvec)
+
+        if ax:
+            rot_x = np.array([[1, 0, 0], [0,
+                                          np.cos(radians), -np.sin(radians)],
+                              [0, np.sin(radians),
+                               np.cos(radians)]])
+            translate_vec = np.array([0, cvec[1], cvec[2]])
+            new_vec = rot_x.dot((vec - translate_vec)) + translate_vec
+
+        if ay:
+            rot_y = np.array([[np.cos(radians), 0,
+                               np.sin(radians)], [0, 1, 0],
+                              [-np.sin(radians), 0,
+                               np.cos(radians)]])
+            translate_vec = np.array([cvec[0], 0, cvec[2]])
+            new_vec = rot_y.dot((vec - translate_vec)) + translate_vec
+
+        if az:
+            rot_z = np.array([[np.cos(radians), -np.sin(radians), 0],
+                              [np.sin(radians),
+                               np.cos(radians), 0], [0, 0, 1]])
+            translate_vec = np.array([cvec[0], cvec[1], 0])
+            new_vec = rot_z.dot((vec - translate_vec)) + translate_vec
+
+        return new_vec
+
+    @staticmethod
+    def angle_elevation(vec: np.ndarray) -> float:
+        """Returns the elevation angle of vector
+
+        Args:
+            vec (np.ndarray): input vector
+
+        Returns:
+            float: elevation angle in radians
+        """
+        unit_z = np.array([0, 0, 1])
+        unit_self = Vec3D.normed(vec)
+        return np.arccos(unit_self.dot(unit_z))
+
+    @staticmethod
+    def angle_azimuth(vec: np.ndarray) -> float:
+        """Returns the azimuthal angle of vector
+
+        Args:
+            vec (np.ndarray): input vector
+
+        Returns:
+            float: azimuthal angle of vector
+        """
+        return np.arctan2(*vec[:2][::-1])
+
+    @staticmethod
+    def two_points_described(points2D: List[np.ndarray]) -> Tuple[np.ndarray]:
+        raise NotImplementedError(
+            "This method does not need to be implemented for 3D.")
+
+    @staticmethod
+    def add_z(vec2D: np.array, z: float = 0.):
+        raise NotImplementedError(
+            "This method does not need to be implemented for 3D.")
+
+    @staticmethod
+    def angle(vector: Vec2D) -> float:
+        raise NotImplementedError(
+            """This method is implemented by 'angle_azimuth' method for 3D vector.
+            Please use that instead.""")
+
+    @staticmethod
+    def rotate_around_point(xy: Vec2D, radians: float,
+                            origin=(0, 0)) -> np.ndarray:
+        raise NotImplementedError(
+            """This method is implemented by 'rotate' method for 3D vector.
+            Please use that instead.""")
+
+    @staticmethod
+    def snap_unit_vector(vec_n: Vec2D, flip: bool = False) -> Vec2D:
```

Comment:
Why would we not want to snaps to either the x,y, or z unit vectors?

Reply:
Oh this was a mistake. Will fix this and add tests for it.